internal/libintl.cpp

```diff
@@ -58,8 +59,19 @@ libintl_lite_bool_t loadMessageCatalog(const char* domain, const char* moFilePat
 
 	FILE* moFile = NULL;
 	CloseFileHandleGuard closeFileHandleGuard(moFile);
-	moFile = fopen(moFilePath, "rb");
-	
+
+	string moFilePathStr = moFilePath;
+	for (const string &path : buildMoFilePaths(domain))
+	{
+		string fullPath = moFilePathStr + path;
+		error_code error;
+		if (filesystem::is_regular_file(fullPath, error))
```

Comment:
I think we can avoid that and directly fopen the file. If the result is not null then break, else continue. No need to go in details about the type of the file.

Reply:
I don't know if I've got something wrong, but if I do

```cpp
FILE *file = fopen("./some_dir", "rb");
int chr = fgetc(file);
```

`file` would be some valid pointer, and `chr` would be `-1`, which probably means the "directory" can be opened, but read failed?

Reply:
I think you are right. The rationale is that since errors are not actually handled we may as well keep going and let `loadMessageCatalogFile()` handle the failure, as-if trying to read a directory was as problematic than trying to read a malformed .mo file. The result would be the same but the algorithm would be simpler, and we save a dependency to the not-yet-so-available c++17. What do you think?

Reply:
I think that's a good idea. I've did the refactor in ed889f7, could you please check them out?